tests/test_requestparameter.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python
+#
+# A library that provides a Python interface to the Telegram Bot API
+# Copyright (C) 2015-2022
+# Leandro Toledo de Souza <devs@python-telegram-bot.org>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser Public License for more details.
+#
+# You should have received a copy of the GNU Lesser Public License
+# along with this program.  If not, see [http://www.gnu.org/licenses/].
+import datetime
+
+import pytest
+
+from telegram import InputFile, MessageEntity, InputMediaPhoto, InputMediaVideo
+from telegram.constants import ChatType
+from telegram.request._requestparameter import RequestParameter
+from tests.conftest import data_file
+
+
+class TestRequestParameter:
+    def test_init(self):
+        request_parameter = RequestParameter('name', 'value', [1, 2])
+        assert request_parameter.name == 'name'
+        assert request_parameter.value == 'value'
+        assert request_parameter.input_files == [1, 2]
+
+        request_parameter = RequestParameter('name', 'value', None)
+        assert request_parameter.name == 'name'
+        assert request_parameter.value == 'value'
+        assert request_parameter.input_files is None
+
+    def test_slot_behaviour(self, mro_slots):
+        inst = RequestParameter('name', 'value', [1, 2])
+        for attr in inst.__slots__:
+            assert getattr(inst, attr, 'err') != 'err', f"got extra slot '{attr}'"
+        assert len(mro_slots(inst)) == len(set(mro_slots(inst))), "duplicate slot"
+
+    @pytest.mark.parametrize(
+        'value, expected',
+        [
+            (1, '1'),
+            ('one', 'one'),
+            (True, 'true'),
+            (None, 'null'),
+            ([1, '1'], '[1, "1"]'),
+            ({True: None}, '{"true": null}'),
```

Comment:
these two tests fail for me:

<details>
<summary>Test result of both:</summary>

```
self = <tests.test_requestparameter.TestRequestParameter object at 0x7f793dc40f40>
value = [1, '1'], expected = '[1, "1"]'

    @pytest.mark.parametrize(
        'value, expected',
        [
            (1, '1'),
            ('one', 'one'),
            (True, 'true'),
            (None, 'null'),
            ([1, '1'], '[1, "1"]'),
            ({True: None}, '{"true": null}'),
            ((1,), '[1]'),
        ],
    )
    def test_json_value(self, value, expected):
        request_parameter = RequestParameter('name', value, None)
>       assert request_parameter.json_value == expected
E       assert '[1,"1"]' == '[1, "1"]'
E         - [1, "1"]
E         ?    -
E         + [1,"1"]

tests/test_requestparameter.py:61: AssertionError

./tests/test_requestparameter.py::TestRequestParameter::test_json_value[value5-{"true": null}] Failed: [undefined]assert '{"True":null}' == '{"true": null}'
  - {"true": null}
  ?   ^     -
  + {"True":null}
  ?   ^
self = <tests.test_requestparameter.TestRequestParameter object at 0x7f793dc40fd0>
value = {True: None}, expected = '{"true": null}'

    @pytest.mark.parametrize(
        'value, expected',
        [
            (1, '1'),
            ('one', 'one'),
            (True, 'true'),
            (None, 'null'),
            ([1, '1'], '[1, "1"]'),
            ({True: None}, '{"true": null}'),
            ((1,), '[1]'),
        ],
    )
    def test_json_value(self, value, expected):
        request_parameter = RequestParameter('name', value, None)
>       assert request_parameter.json_value == expected
E       assert '{"True":null}' == '{"true": null}'
E         - {"true": null}
E         ?   ^     -
E         + {"True":null}
E         ?   ^
```

</details>

Reply:
mh … which machine/os/py version are you on? I can live with the whitespace differences, but `true` vs `True` is strange. AFAIK only the lower case variant is proper json

Reply:
python 3.8.10 on linux, with pytest 7.0.x. I'll try downgrading pytest and see if it fixes it..

Reply:
found the culprit.. it was `ujson`. Uninstalling it made it pass.

Reply:
ah okay. I'll leave as is for now then. we may have to adjust/expand tests anyway once we switch to orjson